lib/discoverable.coffee

```diff
@@ -86,7 +92,40 @@ retrieveServices = ->
 		.return(services)
 
 # Set the services function as a memoized one.
-services = _.memoize(retrieveServices)
+registryServices = _.memoize(retrieveServices)
+
+###
+# @summary Determines if a service is valid.
+# @function
+# @private
+###
+findValidService = (serviceIdentifier, knownServices) ->
+	_.find knownServices, (service) ->
```

Comment:
nit: This could be

```coffeescript
_.find knownServices, ({ service, tags }) ->
  serviceIdentifier is service or serviceIdentifier in tags
```

or even

```coffeescript
_.find knownServices, ({ service, tags }) ->
  serviceIdentifier in [ service, tags... ]
```